finat/mixed.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, print_function, division
+from six import iteritems
+from six.moves import zip
+
+import numpy
+
+import gem
+
+from finat.finiteelementbase import FiniteElementBase
+from finat.enriched import EnrichedElement
+
+
+def MixedElement(elements):
+    """Constructor function for FEniCS-style mixed elements.
+
+    Implements mixed element using :py:class:`EnrichedElement` and
+    value shape transformations with :py:class:`MixedSubElement`.
+    """
+    sizes = [numpy.prod(element.value_shape, dtype=int)
+             for element in elements]
+    offsets = [int(offset) for offset in numpy.cumsum([0] + sizes)]
```

Comment:
Yes, I am adopting this obsessive casting, because in Python 3 `numpy.int64` is not instance of `int`, which sometimes cause annoyances.